### PR TITLE
Minimax and alpha beta pruning.

### LINKED ISSUE #8: Alpha beta minimax.
Implement minimax and basic alpha beta searching for the ai.

### PR DESCRIPTION
**Pull request by benjazzy**
Switched the search algorithm to a minimax search and added alpha beta pruning to speed up the search.  There is still more work to be done with regards to multi threading the alpha beta pruning.  Currently each thread has it's own alpha and beta however it should speed things up considerably to implement alpha beta at the top level.

### CODE REVIEW

**Comment by benjazzy** [comment]
Also fixed an error with user input on windows.

File: src/bin/test/main.rs [hunk 1]
```diff
@@ -63,20 +74,17 @@ fn update(board: Box<Board>, boards: Arc<Vec<BitBoard>>, game: &mut Game) -> (St
 
     //Uncomment for user input on White.
```
Comment (alecrm) [inline src/bin/test/main.rs:75]: Is the end goal to have both options available? AI vs. AI and Human vs AI? Or is the AI vs. AI available right now just for testing?

Comment (benjazzy) [reply src/bin/test/main.rs:75]: The end goal would be to have the option of the ai playing against itself, the ai playing against a human, and the ai playing against a different ai.  I haven't gotten around to implementing a graceful way to switch between ai v ai and ai v human so for now I have just been commenting and uncommenting that code.

Comment (alecrm) [reply src/bin/test/main.rs:75]: Sounds good!

**Review by alecrm — APPROVED** [review]

**Pull request merged**